sys/ecc/golay2412.c

```diff
@@ -0,0 +1,416 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/**
+ * @ingroup     sys_ecc
+ * @{
+ *
+ * @file
+ * @brief       Golay(24,12) half-rate forward error-correction code
+ *
+ * Code referring to Liquid-DSP
+ * https://github.com/jgaeddert/liquid-dsp/blob/master/src/fec/src/fec_golay2412.c
+ *
+ * @author      Joseph Gaeddert
+ * @author      Peter Kietzmann <peter.kietzmann@haw.hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <errno.h>
+
+#include "bitarithm.h"
+#include "ecc/golay2412.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#if ENABLE_DEBUG
+#define DEBUG_FEC_GOLAY2412 (1)
+#else
+#define DEBUG_FEC_GOLAY2412 (0)
+#endif
+
+/* generator matrix transposed [24 x 12] */
+static const unsigned int golay2412_Gt[24] = {
+    0x08ed, 0x01db, 0x03b5, 0x0769, 0x0ed1, 0x0da3, 0x0b47, 0x068f,
+    0x0d1d, 0x0a3b, 0x0477, 0x0ffe, 0x0800, 0x0400, 0x0200, 0x0100,
+    0x0080, 0x0040, 0x0020, 0x0010, 0x0008, 0x0004, 0x0002, 0x0001};
+
+/* P matrix [12 x 12] */
+static const unsigned int golay2412_P[12] = {
+    0x08ed, 0x01db, 0x03b5, 0x0769,
+    0x0ed1, 0x0da3, 0x0b47, 0x068f,
+    0x0d1d, 0x0a3b, 0x0477, 0x0ffe};
+
+/* parity check matrix [12 x 24] */
+static const unsigned int golay2412_H[12] = {
+    0x008008ed, 0x004001db, 0x002003b5, 0x00100769,
+    0x00080ed1, 0x00040da3, 0x00020b47, 0x0001068f,
+    0x00008d1d, 0x00004a3b, 0x00002477, 0x00001ffe};
+
+#if DEBUG_FEC_GOLAY2412
+/* print string of bits to standard output */
+static inline void liquid_print_bitstring(unsigned int _x,
+                            unsigned int _n)
+{
+    unsigned int i;
+    for (i=0; i<_n; i++)
+        printf("%1u", (_x >> (_n-i-1)) & 1);
+}
+#endif
+
+static inline unsigned int block_get_enc_msg_len(unsigned int _dec_msg_len,
+                                       unsigned int _m,
+                                       unsigned int _k)
+{
+    if (_m == 0) {
+        DEBUG("block_get_enc_msg_len(), input block size cannot be zero\n");
+        return ENOTSUP;
+    }
+    else if (_k < _m) {
+        DEBUG("block_get_enc_msg_len(), output block size cannot be smaller than input\n");
+        return ENOTSUP;
+    }
+
+    /* compute total number of bits in decoded message  */
+    unsigned int num_bits_in = _dec_msg_len*8;
+
+    /* compute total number of blocks: ceil(num_bits_in/_m) */
+    unsigned int num_blocks = num_bits_in / _m + (num_bits_in%_m ? 1 : 0);
```

Comment:
add spaces around module `%` and also put parentheses around ternary condition, also see Codacy report.